src/wasm/wasm-type.cpp

```diff
@@ -390,6 +391,71 @@ bool Type::isRtt() const {
   }
 }
 
+void Type::walk(std::function<void(const Type&)> noteRecursion,
+                std::function<void(const Type&)> visitType) {
+  std::set<Type> scanned;
+  std::vector<Type*> scanList = {this};
+  std::vector<Type*> visitList;
```

Comment:
why use `Type*` for these? would it not work with `Type`?

Reply:
Ah, I adapted this from some code that needs to be able to modify the types in-place as they are visited, so I needed to reference the types' original locations. If you won't need this very soon, I'll probably make some changes to share code between the mutable and immutable versions and just include this as part of a larger PR.

Reply:
Take your time, I have other stuff to work on meanwhile then. I'll delay writing of GC stuff and focus on reading.